hyperspy/tests/utils/test_stack.py

```diff
@@ -41,17 +41,23 @@ def test_stack_of_stack(self):
 
     def test_stack_not_default(self):
         s = self.signal
-        s1 = s.deepcopy() + 1
-        s2 = s.deepcopy() * 4
+        s1 = s.inav[:, :-1] + 1
+        s2 = s.inav[:, ::2] * 4
         result_signal = utils.stack([s, s1, s2], axis=1)
         axis_size = s.axes_manager[1].size
+        axs1 = s1.axes_manager[1].size
+        axs2 = s2.axes_manager[1].size
         result_list = result_signal.split()
         assert len(result_list) == 3
-        np.testing.assert_array_almost_equal(
-            result_list[0].data, result_signal.inav[:, :axis_size].data)
-        result_signal = utils.stack([s, s1, s2], axis='y')
-        np.testing.assert_array_almost_equal(
-            result_list[0].data, result_signal.inav[:, :axis_size].data)
+        for rs in [result_signal, utils.stack([s, s1, s2], axis='y')]:
+            np.testing.assert_array_almost_equal(
```

Comment:
In this particular case - I think so. Both stacks should be identical, and the multiple asserts just check that each of the stacked pieces are actually inside (so just testing the same function, but different parts of its result)